keps/sig-node/1029-ephemeral-storage-quotas/README.md

```diff
@@ -709,6 +732,143 @@ and are not reported here.
 | du after umount/mount    | 66.0      | 82.4  | 29.2         | 28.1   |
 | Remove Files             | 188.6     | 156.6 | 90.4         | 81.8   |
 
+## Production Readiness Review Questionnaire
+
+<!--
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: LocalStorageCapacityIsolationFSQuotaMonitoring
+  - Components depending on the feature gate: kubelet\apiserver\controller-manager\scheduler
+
+###### Does enabling the feature change any default behavior?
+
+None. Behavior will not change.
+When LocalStorageCapacityIsolation is enabled for local ephemeral storage and the backing filesystem for emptyDir volumes supports project quotas and they are enabled, use project quotas to monitor emptyDir volume storage consumption rather than filesystem walk for better performance and accuracy.
+
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Set the feature gate to false.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Performance changes.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, test/e2e_node/quota_lsci_test.go
+
+### Rollout, Upgrade and Rollback Planning
+
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+None.
+
+###### What specific metrics should inform a rollback?
+
+None. User can focus on volume metrics.
```

Comment:
It seems like cluster-admins probably want to be able to tell when pods start failing because they run out of space.

Reply:
To see failures, I think we can read kubelet log for eviction related logs or using `xfs_quota` to check the quota settings.

here's an example of volume1048581 (empty dir).
```
[root@daocloud ~]# xfs_quota -x -c 'report -h' /dev/sdc
Project quota on /var/lib/kubelet (/dev/sdc)
                        Blocks
Project ID   Used   Soft   Hard Warn/Grace
---------- ---------------------------------
#0           156K      0      0  00 [------]
volume1048577    16K      0      0  00 [------]
volume1048578     8K      0      0  00 [------]
volume1048579     8K      0      0  00 [------]
volume1048581      0     8E     8E  00 [------]
volume1048582     8K      0      0  00 [------]
volume1048583     8K      0      0  00 [------]
volume1048584      0     8E     8E  00 [------]
[root@daocloud ~]# docker ps | grep test
2e848e2ae430   k8s.gcr.io/test-webserver   "/test-webserver"        4 seconds ago   Up 3 seconds             k8s_test-container_test-pd-1_default_b6fcd777-ffef-4148-ba9a-fbdf4e7cfc8d_0
19281193d915   k8s.gcr.io/test-webserver   "/test-webserver"        6 seconds ago   Up 5 seconds             k8s_test-container_test-pd-2_default_ce4c955b-a04a-4b98-8454-d904ba6341eb_0
57f96f8680eb   k8s.gcr.io/pause:3.2        "/pause"                 9 seconds ago   Up 8 seconds             k8s_POD_test-pd-2_default_ce4c955b-a04a-4b98-8454-d904ba6341eb_0
b14274ba66e8   k8s.gcr.io/pause:3.2        "/pause"                 9 seconds ago   Up 8 seconds             k8s_POD_test-pd-1_default_b6fcd777-ffef-4148-ba9a-fbdf4e7cfc8d_0
[root@daocloud ~]# docker cp data 2e848e2ae430:/cache/data/
[root@daocloud ~]# xfs_quota -x -c 'report -h' /dev/sdc
Project quota on /var/lib/kubelet (/dev/sdc)
                        Blocks
Project ID   Used   Soft   Hard Warn/Grace
---------- ---------------------------------
#0           160K      0      0  00 [------]
volume1048577    16K      0      0  00 [------]
volume1048578     8K      0      0  00 [------]
volume1048579     8K      0      0  00 [------]
volume1048580      0     8E     8E  00 [------]
volume1048581   2.6G     8E     8E  00 [------]
volume1048582     8K      0      0  00 [------]
volume1048583     8K      0      0  00 [------]
volume1048584      0     8E     8E  00 [------]
```